core/templates/dev/head/galleries/Gallery.js

```diff
@@ -201,6 +204,115 @@ oppia.controller('Gallery', [
       }
 
       $rootScope.loadingMessage = '';
+
+      // The following initializes the gallery carousel(s).
+      // Pause is necessary to ensure all elements have loaded.
+      $timeout(initCarousels, 500);
+
+      if (typeof $scope.galleryGroups != 'undefined') {
+        // The following initializes the tracker to have all
+        // elements flush left.
+        cleanLoad = false;
+        for (i = 0; i < $scope.galleryGroups.length; i++) {
+          $scope.rowIndices[i] = 0;
+        }
+      }
+    });
+
+    // Below is the width of each tile.
+    var TILE_DISPLAY_WIDTH = 211.953;
+    // If the value below is changed, the following CSS values in oppia.css
+    // must be changed:
+    // - .oppia-gallery-tiles-carousel: max-width
+    // - .oppia-gallery-tiles-container: max-width
+    var MAX_TILE_DISPLAY = 4;
+    $scope.tileDisplayCount = 0;
+    $scope.galleryTilesResizedWidth = 0;
+
+    var initCarousels = function() {
+      var windowWidth = $(window).width() * 0.85;
+      $scope.tileDisplayCount = Math.floor(windowWidth / TILE_DISPLAY_WIDTH);
+      $scope.galleryTilesResizedWidth = $scope.tileDisplayCount *
+        TILE_DISPLAY_WIDTH;
+      if ($scope.galleryTilesResizedWidth >
+          TILE_DISPLAY_WIDTH * MAX_TILE_DISPLAY) {
+        $scope.galleryTilesResizedWidth = TILE_DISPLAY_WIDTH * MAX_TILE_DISPLAY;
+        $scope.tileDisplayCount = MAX_TILE_DISPLAY;
+      }
+      $('.oppia-gallery-tiles-carousel').css({
+        width: $scope.galleryTilesResizedWidth + 'px'
+      });
+
+      // The following determines whether to enable left scroll after resize.
+      for (i = 0; i < $('.oppia-gallery-tiles-carousel').length; i++) {
+        var carouselJQuerySelector = '.oppia-gallery-tiles:eq(n)'.replace(
+          'n', i);
+        var carouselScrollPositionPx = $(carouselJQuerySelector).scrollLeft();
+        var index = Math.ceil(carouselScrollPositionPx / TILE_DISPLAY_WIDTH);
+        $scope.rowIndices[i] = index;
+      }
+    };
+
+    $scope.scroll = function(ind, isLeftScroll) {
+      var carouselJQuerySelector = '.oppia-gallery-tiles:eq(n)'.replace(
+        'n', ind);
+      var lftOverlaySelector =
+        '.oppia-gallery-tiles-carousel-overlay-left:eq(n)'.replace('n', ind);
+      var rtOverlaySelector =
+        '.oppia-gallery-tiles-carousel-overlay-right:eq(n)'.replace('n', ind);
+
+      var direction = isLeftScroll ? -1 : 1;
+      var carouselScrollPositionPx = $(carouselJQuerySelector).scrollLeft();
+
+      // Scrolling eligibility checks.
+      if ($scope.galleryGroups[ind].activity_summary_dicts.length <=
+          $scope.tileDisplayCount) {
+        return;
+      } else if (isLeftScroll && $scope.rowIndices[ind] == 0) {
```

Comment:
Do you need this (and the following clause)? It seems to be taken care of in the bodies of the if/else statements on line 282 and 285.